core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp

```diff
@@ -1165,26 +1173,137 @@ KOKKOS_INLINE_FUNCTION void parallel_for(
  * The range i=0..N-1 is mapped to all threads of the the calling thread team
  * and a summation of val is performed and put into result.
  */
+
 template <typename iType, class Lambda, typename ValueType>
-KOKKOS_INLINE_FUNCTION void parallel_reduce(
-    const Impl::TeamThreadRangeBoundariesStruct<
-        iType, Impl::OpenMPTargetExecTeamMember>& loop_boundaries,
-    const Lambda& lambda, ValueType& result) {
-  ValueType* tmp_scratch =
+KOKKOS_INLINE_FUNCTION
+    typename std::enable_if<!Kokkos::is_reducer_type<ValueType>::value>::type
+    parallel_reduce(
```

Comment:
Can't you implement this one in terms of the `reducer` implementation using `Kokkos::Sum`?